iac-common/src/test/java/org/sonar/iac/common/testing/Verifier.java

```diff
@@ -162,7 +162,7 @@ public static class TestContext extends TreeContext implements InitContext, Chec
     private final SingleFileVerifier verifier;
     private final List<Issue> raisedIssues = new ArrayList<>();
 
-    public TestContext(SingleFileVerifier verifier) {
+    public TestContext(@Nullable SingleFileVerifier verifier) {
```

Comment:
IMO `TestContext` is strongly coupled with `SingleFileVerifier` and `SingleFileVerifier` is strongly coupled with a file on disk, so such hacks are needed.
Yesterday I also needed to deal with it, see new `verify(TreeParser<Tree> parser, String content, IacCheck check, Issue... expectedIssues)` method on master in this class.
Anyway, let's handle it in a separate story.